### PR TITLE
ci(e2e): Migrate tests to Jest

### PR DESCRIPTION
**Pull request by scinos**
## Changes proposed in this Pull Request

Migrate Calypso e2e tests to Jest.

### Main differences

Jest and Mocha API are very similar. There are a few differences though:

- Jest don't have `this` inside the test. We can't store values there (eg: no more `this.driver = ...`).
- Hooks don't have names.
- `before`/`after` hooks are named `beforeAll`/`afterAll`
- Use `jest.setTimeout()` instead of `this.timeout()` to set test timeouts
- For hooks, the timeout is passed as the second parameter

### Pros:

- Seems to be a bit faster. I got a build in `3m:10s`, while `trunk` is about `3m:30s`. This doesn't even use [Jest caching](https://jestjs.io/docs/configuration#cachedirectory-string), so it may end up running in less than 3 minutes.
- [`jest-teamcity` reporter](https://www.npmjs.com/package/jest-teamcity) adds the test filename as part of the test name in TeamCity. This can help identify which tests is failing:
image

- Log are much easier to read, even when running multiple tests in parallel. It is easy to see what was the test suite, what step failed, the error and the stack trace:
image

### Cons:

- [`jest-teamcity` reporter](https://www.npmjs.com/package/jest-teamcity) adds the test filename as part of the test name... which means that renaming a file will erase the history of the test in TeamCity, as it will appear as a new test.

- Test that are in `specs/specs-calypso` but don't have the `@parallel` tag will appear as disabled in TC UI.

### CODE REVIEW

File: test/e2e/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+module.exports = {
+	env: {
+		node: true,
+		mocha: false,
+	},
+	overrides: [
+		{
+			plugins: [ 'mocha' ],
+			files: [
+				'test/e2e/specs/specs-jetpack/*',
```
Comment (scinos) [inline test/e2e/.eslintrc.js:10]: Enable `mocha` and disable `jest` plugin & rules for these paths

File: test/e2e/lib/hooks/video/video-recorder.js [hunk 2]
```diff
@@ -19,7 +19,6 @@ export const buildHooks = ( displayNum ) => {
 	let ffVideo;
 
 	const startVideoRecording = async () => {
-		console.log( `Start video recording on port :${ displayNum }}` );
```
Comment (scinos) [inline test/e2e/lib/hooks/video/video-recorder.js]: Very noisy

File: test/e2e/lib/jest/environment-fail-fast.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const JestEnvironmentNode = require( 'jest-environment-node' );
```
Comment (scinos) [inline test/e2e/lib/jest/environment-fail-fast.js]: This is used to implement failfast (abort current spec if it fails, don't touch other specs). Inspired by https://github.com/facebook/jest/issues/6527#issuecomment-760092817

**Comment by sarayourfriend** [comment]
The desktop tests failed, is that expected?

**Comment by scinos** [comment]
Abandoned, we migrated Playwright tests instead.

**Pull request closed without merging**